slf4j-jdk-platform-logging/src/test/java/org/slf4j/jdk/SLF4JSystemLoggerTest.java

```diff
@@ -0,0 +1,77 @@
+package org.slf4j.jdk;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.ServiceLoader;
+
+import static org.junit.Assert.*;
+
+public class SLF4JSystemLoggerTest {
+
+    private static final PrintStream ERROR_OUT = System.err;
+    private static final ByteArrayOutputStream OUTPUT = new ByteArrayOutputStream();
+    private static final PrintStream ERROR_OUT_REPLACEMENT = new PrintStream(OUTPUT);
+
+    @Before
+    public void setUp() {
+        System.setErr(ERROR_OUT_REPLACEMENT);
+    }
+
+    @After
+    public void tearDown() {
+        System.setErr(ERROR_OUT);
+    }
+
+    @Test
+    public void loggerFinderLoadedAsOnlyService() {
+        // this method asserts that there is exactly one `LoggerFinder` and its of the correct type
+        getSLF4JSystemLoggerFinder();
+    }
+
+    @Test
+    public void loggerFinderReturnsLogger() {
+        var logger = getSLF4JSystemLogger();
+        assertNotNull("LoggerFinder must return logger", logger);
+    }
+
+    @Test
+    public void loggerLogsMessage() {
+        var logger = getSLF4JSystemLogger();
+        var message = "Test system logging!";
+        logger.log(System.Logger.Level.INFO, message);
+
+        String output = getOutput();
+
+        assertTrue("Captured output should contain logged message.", output.contains(message));
+    }
+
+    private static System.LoggerFinder getSLF4JSystemLoggerFinder() {
+        var loggerFinders = new ArrayList<System.LoggerFinder>();
+        // this fails when test is executed from the module path
+        // because the module declaration does not declare
+        // `uses System.LoggerFinder`
```

Comment:
At the moment, only the class path integration is tested. Module path integration would fail here.

Reply:
I observe the following error:

[ERROR] org.slf4j.jdk.SLF4JSystemLoggerTest.loggerLogsMessage  Time elapsed: 0.003 s  <<< ERROR!
java.util.ServiceConfigurationError: java.lang.System$LoggerFinder: module org.slf4j.jdk does not declare `uses`
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:589)
        at java.base/java.util.ServiceLoader.checkCaller(ServiceLoader.java:575)
        at java.base/java.util.ServiceLoader.<init>(ServiceLoader.java:504)
        at java.base/java.util.ServiceLoader.load(ServiceLoader.java:1687)
        at org.slf4j.jdk@2.0.0-alpha4-SNAPSHOT/org.slf4j.jdk.SLF4JSystemLoggerTest.getSLF4JSystemLoggerFinder(SLF4JSystemLoggerTest.java:63)
        at org.slf4j.jdk@2.0.0-alpha4-SNAPSHOT/org.slf4j.jdk.SLF4JSystemLoggerTest.getSLF4JSystemLogger(SLF4JSystemLoggerTest.java:83)
        at org.slf4j.jdk@2.0.0-alpha4-SNAPSHOT/org.slf4j.jdk.SLF4JSystemLoggerTest.loggerLogsMessage(SLF4JSystemLoggerTest.java:49)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)